core/src/commonMain/kotlin/com/xebia/functional/xef/llm/Embeddings.kt

```diff
@@ -1,8 +1,28 @@
 package com.xebia.functional.xef.llm
 
+import arrow.fx.coroutines.parMap
+import com.xebia.functional.xef.llm.models.embeddings.Embedding
 import com.xebia.functional.xef.llm.models.embeddings.EmbeddingRequest
 import com.xebia.functional.xef.llm.models.embeddings.EmbeddingResult
+import com.xebia.functional.xef.llm.models.embeddings.RequestConfig
 
 interface Embeddings : LLM {
   suspend fun createEmbeddings(request: EmbeddingRequest): EmbeddingResult
+
+  suspend fun embedDocuments(
+    texts: List<String>,
+    requestConfig: RequestConfig,
+    chunkSize: Int?
+  ): List<Embedding> =
+    if (texts.isEmpty()) emptyList()
```

Comment:
I think this way of extracting embeddings which we had since the beginning it's not effitient.
Here we split a chunk of text into the `chunkSize` or `400` and send N request to the open AI embeddings endpoint.

In reality the embeddings endpoint by default looks like this:
- cl100k_base encoding with 8191 max input tokens.

I think it'd be more efficient to split the text based on the model's `max input tokens` and then send fewer and larger requests close to the limit.
Something like:
`chunked(modelType.maxTokens - 100).parMap...`

Reply:
This is really interesting. We should address this in a separate PR 👍